File: 백준/Silver/11723. 집합/집합.py
import sys

m = int(sys.stdin.readline())
lst = [0] * 21

for _ in range(m):
    temp = sys.stdin.readline().strip().split()

    if len(temp) == 1:
        if temp[0] == "all":
            lst = [1] * 21
            lst[0] = 0
        else:
            lst = [0] * 21

    else:
        func, x = temp[0], temp[1]
        x = int(x)

        if func == "add":
            lst[x] = 1
        elif func == "remove":
            lst[x] = 0
        elif func == "check":
            if lst[x] != 0:
                print(1)
            else:
                print(0)
        elif func == "toggle":
            if lst[x] != 0:
                lst[x] = 0
            else:
                lst[x] = 1

File: 백준/Silver/11723. 집합/집합.java

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.*;
import java.io.IOException;

public class Main {

	public static void main(String[] args) throws IOException {
		StringBuilder sb = new StringBuilder();
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int M = Integer.parseInt(br.readLine());
		int mask = 0;
		for (int i = 0; i < M; i++) {
			String str = br.readLine();

			if ("all".equals(str)) {
				mask = (1 << 21) - 2;
				continue;
			} else if ("empty".equals(str)) {
				mask = 0;
				continue;
			}

			StringTokenizer st = new StringTokenizer(str);
			String comm = st.nextToken();
			int num = Integer.parseInt(st.nextToken());
			if ("add".equals(comm)) {
				mask |= (1 << num);
			} else if ("remove".equals(comm)) {
				mask &= ~(1 << num);
			} else if ("check".equals(comm)) {
				boolean flag;
				sb.append((mask & (1 << num)) != 0 ? 1 : 0).append('\n');
			} else if ("toggle".equals(comm)) {
				if ((mask & (1 << num)) != 0) {
					mask &= ~(1 << num);
				} else {
					mask |= (1 << num);
				}
			}
		}
		System.out.println(sb);
	}

}

